### ISSUE TITLE
[$250] Name and avatar are not shown when splitting or requesting with new user 

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#42023]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.72-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @kevinksullivan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1714664700914269

## Action Performed:

1. Split bill from global create
2. Select 2 or 3 users and one of the user with who there is no existing chat 
3. Split the expense
4. Logout and login back
5. Open the chat with the new user where the split was done
## Expected Result:
Name and avatars should be shown
## Actual Result:
Fallback avatar is shown, no default avatar and names
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
![Snip - (9) New Expensify - Google Chrome (2)](https://github.com/Expensify/App/assets/38435837/4beed6bc-8621-491e-949b-f8311b4d18ec)

![image (1)](https://github.com/Expensify/App/assets/38435837/c5a30250-ca88-4ef9-b44f-5250ffd68ab4)

https://github.com/Expensify/App/assets/38435837/43ac294b-20fe-4514-84c9-3f44dc55ea54

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0110a992c6f2181e4c</li>
        <li>Upwork Job ID: 1790125179814068224</li>
        <li>Last Price Increase: 2024-05-13</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @hungvu193</details>

**Comment by youssef-lr** [comment]
This is due to ioUReports not being loaded when opening a report. I think we should get the display names from somewhere else, maybe using the report's participants.

**Comment by abzokhattab** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Name and avatar are not shown when splitting or requesting with new user
### What is the root cause of that problem?
The `iouReport` is not loaded from the DB when logging in and opening the user report page therefore the `iouReport` resolves to `{}`, that why the names and the avatars of the participants are shown as empty 


### What changes do you think we should make in order to solve the problem?

i checked whether we can get the participants info from the `reportParticipants` arr but found that it only contains the account ID of the opened report:
<img width="405" alt="image" src="https://github.com/Expensify/App/assets/59809993/2ca3f599-9197-4f84-a883-2c28899a647d">

therefore i had to check for another way to handle this. 

what we can do is to fetch the `iouReport` only once if the `iouReport` is empty the same way as we are doing [here](https://github.com/Expensify/App/blob/3fbe52e4eaa8d7adf55f9debd8b747c73bbc8f58/src/pages/TransactionReceiptPage.tsx#L40-L47).   to achieve that we need to add the following `useEffect` inside the [ReportActionItem](https://github.com/Expensify/App/blob/5ab6a0fb7e5e1e74951460abf42cac1da36dcd01/src/pages/home/report/ReportActionItem.tsx#L378-L379) 

```js
    useEffect(() => {
        const iouReportID = ReportActionsUtils.getIOUReportIDFromReportActionPreview(action);
        if (iouReportID && (iouReportID === '0' || !isEmptyObject(iouReport))) {
            return;
        }
        Report.openReport(iouReportID);
        // I'm disabling the warning, as it expects to use exhaustive deps, even though we want this useEffect to run only on the first render.
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
```

**Comment by abzokhattab** [comment]
I think its a duplicate of https://github.com/Expensify/App/issues/38702

**Comment by youssef-lr** [comment]
Thanks @abzokhattab, I think we can close this in favor of the older issue.
